src/engine/openspaceengine.cpp

```diff
@@ -1546,6 +1584,144 @@ void OpenSpaceEngine::toggleShutdownMode() {
+                    nav.up = navStateProfile.up;
+                }
+                if (navStateProfile.yaw.has_value()) {
+                    nav.yaw = navStateProfile.yaw.value();
+                }
+                if (navStateProfile.pitch.has_value()) {
+                    nav.pitch = navStateProfile.pitch.value();
+                }
+                global::navigationHandler->setNavigationStateNextFrame(nav);
+            },
+            [](const Profile::CameraGoToGeo& geo) {
+                std::string geoScript = fmt::format("openspace.globebrowsing.goToGeo"
+                    "([[{}]], {}, {}", geo.anchor, geo.latitude, geo.longitude);
+                if (geo.altitude.has_value()) {
+                    geoScript += fmt::format(", {}", geo.altitude.value());
+                }
+                geoScript += ")";
+                global::scriptEngine->queueScript(
+                    geoScript,
+                    scripting::ScriptEngine::RemoteScripting::Yes
+                );
+            }
+        },
+        p.camera.value()
+    );
+}
+
+void OpenSpaceEngine::setFromProfile_deltaTimes(const Profile& p) {
+    global::timeManager->setDeltaTimeSteps(p.deltaTimes);
+}
+
+void OpenSpaceEngine::setFromProfile_modules(const Profile& p)
+{
+    for (Profile::Module mod : p.modules) {
+        const std::vector<OpenSpaceModule*>& m = global::moduleEngine->modules();
+        const auto it = std::find_if(m.begin(), m.end(),
+            [&mod](const OpenSpaceModule* moduleSearch) {
+                return (moduleSearch->identifier().compare(mod.name) == 0);
+            });
+        if (it != m.end()) {
+            if (mod.loadedInstruction.has_value()) {
+                global::scriptEngine->queueScript(
+                    mod.loadedInstruction.value(),
+                    scripting::ScriptEngine::RemoteScripting::Yes
+                );
+            }
+        }
+        else {
+            if (mod.notLoadedInstruction.has_value()) {
+                global::scriptEngine->queueScript(
+                    mod.notLoadedInstruction.value(),
+                    scripting::ScriptEngine::RemoteScripting::Yes
+                );
+            }
+        }
+    }
+}
+
+void OpenSpaceEngine::setFromProfile_actions(const Profile& p)
+{
+    for (Profile::Action a : p.actions) {
+        if (a.identifier.empty()) {
+            LERROR("Identifier must to provided to register action");
+        }
+        if (global::actionManager->hasAction(a.identifier)) {
+            LERROR(fmt::format("Action for identifier '{}' already existed & registered",
+                a.identifier));
+        }
+        if (a.script.empty()) {
+            LERROR(fmt::format("Identifier '{}' doesn't provide a Lua command to execute",
+                a.identifier));
+        }
+        interaction::Action action;
+        action.identifier = a.identifier;
+        action.command = a.script;
+        if (!a.name.empty()) {
```

Comment:
I don't think there is a need to check if it is empty?  Since the action.name, documentation and guipath are empty by default, in the worst case we would overwrite an empty string with an empty string, which is going to be pretty cheap